src/App.tsx

```diff
@@ -14,11 +15,76 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isVisible: true,
+  };
+
+  showGoodsList = () => {
+    this.setState({
+      isVisible: false,
+    });
+  };
+
+  reverseList = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortAlphavet = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState((state) => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  resetList = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  };
+
+  render(): React.ReactNode {
+    const { goods, isVisible } = this.state;
+
+    return (
+      <div className="App">
+        {isVisible ? (
+          <button type="button" onClick={this.showGoodsList} className="btn btn-success">
```

Comment:
If there are 3 or more properties, put each on new line
```suggestion
          <button
          	type="button" 
          	onClick={this.showGoodsList} 
          	className="btn btn-success"
          >
```